app/forms/waste_exemptions_engine/address_lookup_form_base.rb

```diff
@@ -21,19 +18,16 @@ def look_up_addresses
       end
     end
 
-    # If an address has already been assigned to the registration, pre-select it
-    def preselect_existing_address
-      return unless can_preselect_address?
-
-      selected_address = temp_addresses.detect { |address| address["uprn"].to_s == existing_address.uprn }
-      send("#{address_type}_address", selected_address["uprn"]) if selected_address.present?
-    end
+    def get_address_data(uprn, type)
+      return {} if uprn.blank?
 
-    def can_preselect_address?
-      return false unless existing_address
-      return false unless existing_address.uprn.present?
+      data = temp_addresses.detect { |address| address["uprn"] == uprn.to_i }
+      return {} unless data
 
-      true
+      data
+        .except("address")
+        .except("state_date")
+        .merge(address_type: type, mode: :lookup)
```

Comment:
I have changed the strategy here slightly so that the `uprn` is coming from the form as per standard-rails attribute and then is used to get the data as per before. The attributes are then sent to rails as normal strong parameters for mass-assignment